00-programming-fundamentals/08-intro-to-automated-tests.md

```diff
@@ -60,24 +66,273 @@ true
 
-### Assertions
-An _assertion_ is the part of the _automated test suite_ that tells us whether our code did what we expected it to. Usually, _assertions_ are special methods that will evaluate whether an expression is `true`. If it is, the _assertion_ is true, the test passes (green test), otherwise it fails (red test).
 
-### Exercise: Write an `assert` method
-Let's build our own `assert` method in the `exponate_test.rb` file. Here's what it should look like:
+## Now Comes Minitest
+
+Writing your own tests with puts is wonderful, but it would be handy to have a standard way that developers can use to write tests on their code, a way that other developers understand.  The maintainers of the Ruby language have adopted a testing library called [Minitest](http://docs.seattlerb.org/minitest/) as the default standard for testing in Ruby & later Rails.  For the remainder of your time using Ruby at Ada, we will be using Minitest to write unit-tests for your code.  There are other testing frameworks, notably [RSpec](http://rspec.info/	), but the key concepts are similar.  
+
+
+
+### How To Use Minitest?
+
+Lets say we need to write a class which simulates a 6-sided die.  We could use such a module in any number of board games (Monopoly etc).  
+
+To start with we'll create a file to test the class.  We'll call it `die_test.rb`.  We will also require `minitest/autorun`
+
+```ruby
+# die_test.rb
+
+require 'minitest/autorun'
+require 'minitest/reporters'
+require_relative 'die'
+```
+
+Then we can create an empty file `die.rb` which will hold our class.  
+
+```ruby
+# die.rb
+```
+
+We can run the tests by typing:  `ruby die_test.rb` and get the following:
+
+```bash
+$ ruby die_test.rb
+Run options: --seed 35264
+
+# Running:
+
+
+
+Finished in 0.000751s, 0.0000 runs/s, 0.0000 assertions/s.
+
+0 runs, 0 assertions, 0 failures, 0 errors, 0 skips
+```
+
+We are now setup, but we haven't actually written a test yet.
+
+
+### A Tale of Two Styles
+
+In the TDD World there are two styles of testing.  In the first more traditional method people use *assertions* which are statements that check if a value is what it should be.  The other method is a subset of TDD, called Behavior-Driven Development (BDD) which accomplishes the same thing in a more English-friendly fashion that even non-developers can understand. 
+
+
+#### Writing Your First Test - Assertion Style
+
+To start we'll need to create a TestCase class.  We create a set of unit tests by subclassing MiniTest::Unit::TestCase and add each set of tests in instance methods.  
+
+
+```
+# die_test.rb
+
+require 'minitest/autorun'
+require 'minitest/reporters'
+require_relative 'die'
+
+class TestDie < MiniTest::Unit::TestCase
+  def test_creation_of_die
+    @die = Die.new
```

Comment:
I think this should probably be a local variable instead of an instance variable at this point